test/common/upstream/health_checker_impl_test.cc

```diff
@@ -785,6 +815,29 @@ TEST_F(TcpHealthCheckerImplTest, Timeout) {
   cluster_->runCallbacks({}, removed);
 }
 
+TEST_F(TcpHealthCheckerImplTest, NoData) {
+  InSequence s;
+
+  setupNoData();
+  cluster_->hosts_ = {HostSharedPtr{new HostImpl(
+      cluster_->info_, "", Network::Utility::resolveUrl("tcp://127.0.0.1:80"), false, 1, "")}};
```

Comment:
@hennna might have some insight into how things like this should work to support IPv6 only test environments.